### PR TITLE
fix searchResult next

### PR DESCRIPTION
**Pull request by benoitvidis**
## What does this PR do?

This PR attempts to fix 2 small bugs in the `SearchResult.next` method

1. [from, size] If the from value is set to 0, the next won't return any further page
2. [sort, size] Only the top level entries of the sort options would be taken by the [`search_after`](https://www.elastic.co/guide/en/elasticsearch/reference/master/search-request-search-after.html) call

### Other changes

I had to update some dependencies for babel to make it build again.

### CODE REVIEW

File: src/controllers/searchResult/base.js [hunk 1]
```diff
@@ -54,12 +55,14 @@ class SearchResultBase {
...
-        if (typeof sort === 'string') {
-          request.search_after.push(hit._source[sort]);
-        }
-        else {
-          request.search_after.push(hit._source[Object.keys(sort)[0]]);
-        }
+        const key = typeof sort === 'string'
+          ? sort
+          : Object.keys(sort)[0];
+        const value = key === '_id'
```
Comment (benoitvidis) [inline src/controllers/searchResult/base.js]: _id is a valid sort field and it not part of the _source

File: package.json [hunk 2]
```diff
@@ -27,11 +27,12 @@
   "main": "index.js",
   "license": "Apache-2.0",
   "dependencies": {
+    "lodash.get": "^4.4.2",
```
Comment (Aschen) [inline package.json]: This dependency add 4.5Kb to the bundle (https://bundlephobia.com/result?p=lodash.get@4.4.2).
Maybe you could just add a function like this one https://gist.github.com/Aschen/6ba04c9eca3051a83f7e1dbddf38c6a1

File: src/controllers/searchResult/base.js [hunk 3]
```diff
@@ -54,12 +55,14 @@ class SearchResultBase {
...
-          request.search_after.push(hit._source[sort]);
-        }
-        else {
-          request.search_after.push(hit._source[Object.keys(sort)[0]]);
-        }
+        const key = typeof sort === 'string'
+          ? sort
+          : Object.keys(sort)[0];
+        const value = key === '_uid'
+          ? hit._id
```
Comment (ballinette) [inline src/controllers/searchResult/base.js]: according to ES documentation, I understand that if results are sorted by `_uid`, the "search_after" item whould be `this._request.collection + '#" + hit._id` instead (see https://www.elastic.co/guide/en/elasticsearch/reference/5.6/search-request-search-after.html )
and we also should consider sorting by `_id`

Comment (benoitvidis) [reply src/controllers/searchResult/base.js]: :+1:  good catch

File: package.json (hunk 2 above)
Comment (benoitvidis) [reply package.json]: :+1:

**Pull request merged**